File: 프로그래머스 Lv.1/신고결과 받기.py
def solution(id_list, report, k):
    r=set()
    data=dict()
    for i in report:
        a,b= i.split()
        r.add((a,b))
    for i in r:
        if i[1] in data.keys():
            data[i[1]]+=1
        else:
            data[i[1]]=1
    ans=dict()
    for i in id_list:
        ans[i]=0
    for i in r:
        if i[1] in data.keys() and data[i[1]]>=k: 
            ans[i[0]]+=1
    answer=[]        
    for i in id_list:
        answer.append(ans[i])
    return answer

File: 자바/신고 결과 받기.java
 
import java.util.*;
class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
         
        HashMap<String,Integer> map = new HashMap<>(); 
        HashMap<String,Integer> map2 = new HashMap<>();
        for (int i=0;i<id_list.length;i++){
            map.put(id_list[i],0);
            map2.put(id_list[i],0);
        }
        Set<String> set = new HashSet<String>();       
        for (int i=0;i<report.length;i++){
            set.add(report[i]);
        }
        Iterator<String> it = set.iterator();
        while (it.hasNext()){
            String combi=it.next();
            String [] users=combi.split(" ");
            int before=map.get(users[1]);
            map.put(users[1],before+1);
                
        }
        it = set.iterator();
        while (it.hasNext()){
            String combi=it.next();
            String [] users=combi.split(" ");
            if (map.get(users[1])>=k){
                int before=map2.get(users[0]);
                map2.put(users[0],before+1);
            }
        }
        for (int i=0;i<id_list.length;i++){
            answer[i]=map2.get(id_list[i]);
        }    

         
        return answer;
    }
}
